python/cuml/common/input_utils.py

```diff
@@ -142,6 +153,11 @@ def input_to_cuml_array(X, order='F', deepcopy=False,
 
     elif hasattr(X, "__array_interface__") or \
             hasattr(X, "__cuda_array_interface__"):
+
+        if force_contiguous or hasattr(X, "__array_interface__"):
+            if not _check_array_contiguity(X):
+                X = cp.array(X, order=order, copy=True)
```

Comment:
Maybe issue a debug-level notice that we are making an extra copy for contiguity?